carbon-feature-plugin/src/main/java/org/wso2/maven/p2/feature/install/FeatureInstallMojo.java

```diff
@@ -111,6 +129,28 @@ private FeatureInstaller constructFeatureInstaller() {
         resourceBundle.setLauncher(this.launcher);
         resourceBundle.setForkedProcessTimeoutInSeconds(this.forkedProcessTimeoutInSeconds);
         resourceBundle.setLog(getLog());
+        resourceBundle.setRuntimeLocation(this.runtimeLocation);
         return new FeatureInstaller(resourceBundle);
     }
+
+    /**
+     * Runtime in which the director application is executed.
+     * standalone (default) - to create and use a stand-alone installation of the director application. This option is
+     * needed if the product to be installed includes artifacts with meta-requirements (e.g. to a non-standard
+     * touchpoint action).
+     *
+     * @throws MojoFailureException
+     * @throws MojoExecutionException
+     */
+    private void setDirectorRuntime() throws MojoFailureException, MojoExecutionException {
+        switch (directorRuntime) {
```

Comment:
Currently the plugin only supports "standalone" runtime. In tycho, it provides both "standalone" and "osgi" runtimes. The switch case is added so that this method can be enhanced to support both runtimes based on the configuration.